pkg/inframanager/store/storepolicy.go

```diff
@@ -0,0 +1,342 @@
+		if err != nil {
+			log.Error("Error unmarshalling data from ", WorkerepFile, err)
+			return false
+		}
+		return true
+	}
+
+}
+
+func (policyadd Policy) WriteToStore() bool {
+	PolicySet.PolicyLock.Lock()
+	PolicySet.PolicyMap[policyadd.PolicyName] = policyadd
+	PolicySet.PolicyLock.Unlock()
+	return true
+}
+
+func (ipsetadd IpSet) WriteToStore() bool {
+	PolicySet.PolicyLock.Lock()
+	PolicySet.IpSetMap[ipsetadd.IpsetId] = ipsetadd
+	PolicySet.PolicyLock.Unlock()
+	return true
+}
+
+func (workerepadd PolicyWorkerEndPoint) WriteToStore() bool {
+	PolicySet.PolicyLock.Lock()
+	PolicySet.WorkerEpMap[workerepadd.WorkerEp] = workerepadd
+	PolicySet.PolicyLock.Unlock()
+	return true
+}
+
+func remove(s []string, r string) ([]string, bool) {
+	ret := false
+	for i, v := range s {
+		if v == r {
+			ret = true
+			return append(s[:i], s[i+1:]...), ret
+		}
+		ret = false
+	}
+	return s, ret
+}
+
+func (policydel Policy) DeleteFromStore() bool {
+	//delete the corresponding ipsetid from ipset map
+	var f bool
+	for policyid, _ := range policydel.PolicyId {
+		for ruleid, _ := range policydel.PolicyId[policyid].RuleId {
+			ipsetid := policydel.PolicyId[policyid].RuleId[ruleid].IpSetID
+			if ipsetid != "" {
+				PolicySet.PolicyLock.Lock()
+				delete(PolicySet.IpSetMap, ipsetid)
+				PolicySet.PolicyLock.Unlock()
+			}
+		}
+	}
+	PolicySet.PolicyLock.Lock()
+	delete(PolicySet.PolicyMap, policydel.PolicyName)
+	PolicySet.PolicyLock.Unlock()
+
+	//delete corresponding policy name from worker ep map as well
+	for ep, val := range PolicySet.WorkerEpMap {
+		val.PolicyNameIngress, f = remove(val.PolicyNameIngress, policydel.PolicyName)
+		if f {
+			PolicySet.WorkerEpMap[ep] = val
+		}
+		val.PolicyNameEgress, f = remove(val.PolicyNameEgress, policydel.PolicyName)
+		if f {
+			PolicySet.WorkerEpMap[ep] = val
+		}
+	}
+
+	return true
+}
+
+func (ipsetdel IpSet) DeleteFromStore() bool {
+	policyid := ipsetdel.PolicyId
+	ruleid := ipsetdel.RuleId
+
+	res := PolicySet.PolicyMap[ipsetdel.PolicyName]
+	if reflect.DeepEqual(res, Policy{}) {
+		return false
+	} else {
+		PolicySet.PolicyLock.Lock()
+		if p, ok1 := res.PolicyId[policyid]; ok1 {
+			if r, ok2 := p.RuleId[ruleid]; ok2 {
+				r.IpSetID = ""
+				p.RuleId[ruleid] = r
+			}
+			res.PolicyId[policyid] = p
+		}
+		PolicySet.PolicyLock.Unlock()
+	}
+
+	PolicySet.PolicyLock.Lock()
+	delete(PolicySet.IpSetMap, ipsetdel.IpsetId)
+	PolicySet.PolicyLock.Unlock()
+
+	return true
+}
+
+func (workerepdel PolicyWorkerEndPoint) DeleteFromStore() bool {
+	PolicySet.PolicyLock.Lock()
+	delete(PolicySet.WorkerEpMap, workerepdel.WorkerEp)
+	PolicySet.PolicyLock.Unlock()
+	return true
+}
+
+func (policyget Policy) GetFromStore() store {
+	res := PolicySet.PolicyMap[policyget.PolicyName]
+
+	if reflect.DeepEqual(res, Policy{}) {
+		return nil
+	} else {
+		return res
+	}
+}
+
+func (ipsetget IpSet) GetFromStore() store {
+	res := PolicySet.IpSetMap[ipsetget.IpsetId]
+	if reflect.DeepEqual(res, IpSet{}) {
```

Comment:
DeepEqual returns true or false. (0-false), so if they are equal, it should return true (1)

Reply:
Vishala, the Get method should return the matching IPset when DeepEqual returns true, isn't it?
Can you double check this?